src/ripple/app/paths/PathRequest.cpp

```diff
@@ -408,6 +478,87 @@ void PathRequest::resetLevel (int l)
         iLastLevel = l;
 }
 
+bool
+PathRequest::findPaths (
+    RippleLineCache::ref cache,
+        FindPaths& fp, Issue const& issue,
+            Json::Value& jvArray)
+{
+    STPathSet& spsPaths = mContext[issue];
+    STPath fullLiquidityPath;
+    if (! fp.findPathsForIssue(issue,
+        spsPaths, fullLiquidityPath))
+    {
+        m_journal.debug << iIdentifier << " No paths found";
+        return false;
+    }
+
+    boost::optional<PaymentSandbox> sandbox;
+    sandbox.emplace(&*cache->getLedger(), tapNONE);
+
+    auto& sourceAccount = ! isXRP(issue.account)
+        ? issue.account
+        : isXRP(issue.currency)
+        ? xrpAccount()
+        : *raSrcAccount;
+    STAmount saMaxAmount({ issue.currency, sourceAccount }, 1);
+    saMaxAmount.negate();
+    
+    m_journal.debug << iIdentifier
+        << " Paths found, calling rippleCalc";
+    
+    auto rc = path::RippleCalc::rippleCalculate(
+        *sandbox, saMaxAmount, saDstAmount,
+            *raDstAccount, *raSrcAccount, spsPaths);
+
+    if (! fullLiquidityPath.empty() &&
+        (rc.result() == terNO_LINE || rc.result() == tecPATH_PARTIAL) &&
+        saDstAmount != STAmount(saDstAmount.issue(), 1u, 0, true))
+    {
+        m_journal.debug << iIdentifier
+            << " Trying with an extra path element";
+        spsPaths.push_back(fullLiquidityPath);
+        sandbox.emplace(&*cache->getLedger(), tapNONE);
+        
+        rc = path::RippleCalc::rippleCalculate(
+            *sandbox, saMaxAmount, saDstAmount,
+                *raDstAccount, *raSrcAccount, spsPaths);
+        if (rc.result() != tesSUCCESS)
+        {
+            m_journal.warning << iIdentifier
+                << " Failed with covering path "
+                << transHuman(rc.result());
+        }
+        else
+        {
+            m_journal.debug << iIdentifier
+                << " Extra path element gives "
+                << transHuman(rc.result());
+        }
+    }
+
+    if (rc.result () == tesSUCCESS)
+    {
+        Json::Value jvEntry (Json::objectValue);
+        rc.actualAmountIn.setIssuer (sourceAccount);
+        jvEntry[jss::source_amount] = rc.actualAmountIn.getJson (0);
+        jvEntry[jss::paths_computed] = spsPaths.getJson (0);
+
+        if (hasCompletion ())
+        {
+            // Old ripple_path_find API requires this
+            jvEntry[jss::paths_canonical] = Json::arrayValue;
+        }
+
+        jvArray.append (jvEntry);
+        return true;
+    }
+    
+    m_journal.debug << iIdentifier << " rippleCalc returns "
+        << transHuman (rc.result ());
+    return false;
+}
```

Comment:
While I don't completely understand this function entirely yet, I did want to note for myself and others that I have verified that if faithfully migrates the functionality of the removed block starting at https://github.com/ripple/rippled/pull/1209/files#diff-39a6cdaf81297515fe5f026a03ec2addL515